src/troute-network/troute/DataAssimilation.py

```diff
@@ -687,6 +694,81 @@ def update_for_next_loop(self, network, da_run,):
         if not self.usgs_df.empty:
             self._usgs_df = self.usgs_df.loc[:,network.t0:]
 
+class great_lake(AbstractDA):
+    '''
+    Here is a list of the waterbody IDs and the gage they should correspond to:
+    4800002 -> 04127885 -> 
+    4800004 -> 04159130 -> 13196034 (segment_id)
+    4800006 -> 02HA013
+    4800007 -> IJC file    
+    '''
+    def __init__(self, network, from_files, value_dict, da_run):
+        greatLake = False
+        data_assimilation_parameters = self._data_assimilation_parameters
+        run_parameters = self._run_parameters
+        reservoir_persistence_da = data_assimilation_parameters.get('reservoir_da', None).get('reservoir_persistence_da', None)
+
+        if reservoir_persistence_da:
+            greatLake = reservoir_persistence_da.get('reservoir_persistence_greatLake', False)
+
+        if greatLake:
+
+            streamflow_da_parameters = data_assimilation_parameters.get('streamflow_da', None)
+            
+            if not self._canada_is_created and ('canada_timeslice_files' in da_run):
+                self._canada_df = _create_canada_df(data_assimilation_parameters, streamflow_da_parameters, run_parameters, network, da_run)
+                self._canada_is_created = True
+
+            if 'LakeOntario_outflow' in da_run:
+                self._lake_ontario_df = _create_LakeOntario_df(run_parameters, network, da_run)
+            else:
+                self._lake_ontario_df = pd.DataFrame() 
+            # determine if user explictly requests streamflow DA
```

Comment:
is this comment supposed to be here?

Reply:
The last line here, "determine if user..."

Reply:
deleted